### ISSUE TITLE
unable to test custom repositories

### ISSUE THREAD

**Issue by andreyorlov33** [wovalle/fireorm#232]
Hey! I am attempting to write unit tests for my custom repository layer and **getRepository** is not working as expected. 

`**Booth.ts:**`
`@CustomRepository(Booth)
class BoothRepository extends BaseFirestoreRepository<Booth> {
	async createBooth(props: BoothDto): Promise<Booth> {
		const booth = {
			id: await nanoid(),
			created_at: Date.now(),
			...props,
		};
		return this.create(booth);
	}`
	
`**Booth.spec.ts**`:
`
import { Booth } from '../../collections';
import {initializeAdminApp} from '@firebase/testing'
import { getRepository, initialize } from 'fireorm';
describe(`TEST`, () => {
	let boothRepository: BoothRepository
	beforeEach(()=> {
		const firestore = initializeAdminApp({projectId: 'XXXXXXXXX'}).firestore()
		initialize(firestore)
		boothRepository = getRepository(Booth) as BoothRepository
	})
	it(`should create a Booth and query it`, async () => {
		const walls = {};
		walls[0] = 'test wall 1';
		walls[1] = 'test wall 2';
		const boothDto: BoothDto = {
			description: 'booth used for testing',
			name: 'Test booth',
			roles: {
				user1: BoothRole.OWNER,
			},
			walls,
			// created_at: Date.now(),
			// id: 'foo'
		};
		const b = await boothRepository.createBooth(boothDto);
                 expect(b.name).toBe(boothDto.name)
	});
});
`

Result:
**TypeError: boothRepository.createBooth is not a function**

**Comment by wovalle** [comment]
From what I see `getRepository` is returning a BaseRepository rather than the custom one. For fireorm to recognize that there are CustomRepositories the decorator needs to run first. Are you importing the file where you're declaring your custom repository in your test file?

It says that `Booth.ts` is the file with the custom repository but it might be the model

**Comment by andreyorlov33** [comment]
@wovalle  thank you for your reply.   
Here is how the code is setup ....
**BoothRepository.ts**
```
import {
	BaseFirestoreRepository,
	CustomRepository,
	getRepository,
} from 'fireorm';
import { nanoid } from 'nanoid';
import { Booth } from '../../collections';
import { BoothDto } from './BoothDto';
import { NO_BOOTH_FOUND_ERROR } from './NO_BOOTH_FOUND_ERROR';

@CustomRepository(Booth)
export class BoothRepository extends BaseFirestoreRepository<Booth> {
	async createBooth(props: BoothDto): Promise<Booth> {
		const booth = {
			id: await nanoid(),
			created_at: Date.now(),
			...props,
		};
		return this.create(booth);
	}
}

export const boothRepository = getRepository(Booth) as BoothRepository;
```

**Booth.spec.ts** (as shown in my original post):
```
import { Booth } from '../../collections'; // <-- import collection 
import {initializeAdminApp} from '@firebase/testing'
import { getRepository, initialize } from 'fireorm';
import { BoothRepository } from './Booth'; // <-- BoothRepository Class Import from BoothRepository.ts

describe(`TEST`, () => {
	let boothRepository: BoothRepository
	beforeEach(()=> {
		const firestore = initializeAdminApp({projectId: 'XXXXXXXXX'}).firestore() 
		initialize(firestore) 
		boothRepository = getRepository(Booth) as BoothRepository  // <-- pass collection into getRepository
	})
	it(`should create a Booth and query it`, async () => {
		const walls = {};
		walls[0] = 'test wall 1';
		walls[1] = 'test wall 2';
		const boothDto: BoothDto = {
			description: 'booth used for testing',
			name: 'Test booth',
			roles: {
				user1: BoothRole.OWNER,
			},
			walls,
			// created_at: Date.now(),
			// id: 'foo'
		};
		const b = await boothRepository.createBooth(boothDto);
                 expect(b.name).toBe(boothDto.name)
	});
});
```

**Error** : `    TypeError: boothRepository.createBooth is not a function`

**Comment by leandrogehlen** [comment]
I guess, this issue is happening because collection class is declared in another file, all collection class in this repository are declared in same test file

**Comment by wovalle** [comment]
Yeah, that's a typescript limitation, not a fireorm one. In other repos I [need to force the typescript compiler](https://github.com/wovalle/yaab/blob/master/src/index.ts#L71-L82) to evaluate files so their Decorators are registered, maybe you have to do the same.
